src/Credentials/ExternalAccountCredentials.php

```diff
@@ -276,9 +279,27 @@ public function fetchAuthToken(callable $httpHandler = null)
         return $stsToken;
     }
 
-    public function getCacheKey()
+    /**
+     * Get the cache token key for the credentials.
+     * The cache token key format depends on the type of source
+     * The format for the cache key one of the following:
+     * FetcherCacheKey.Scope.[ServiceAccount].[TokenType].[WorkforcePoolUserProject]
+     * FetcherCacheKey.Audience.[ServiceAccount].[TokenType].[WorkforcePoolUserProject]
+     *
+     * @return ?string;
+     */
+    public function getCacheKey(): ?string
     {
-        return $this->auth->getCacheKey();
+        $scopeOrAudience = $this->auth->getAudience();
+        if (!$scopeOrAudience) {
+            $scopeOrAudience = $this->auth->getScope();
+        }
+
+        return $this->auth->getSubjectTokenFetcher()->getCacheKey() .
+            '.' . $scopeOrAudience .
+            $this->concatenateValueOrEmpty($this->serviceAccountImpersonationUrl) .
+            $this->concatenateValueOrEmpty($this->auth->getSubjectTokenType()) .
+            $this->concatenateValueOrEmpty($this->workforcePoolUserProject);
```

Comment:
with consistency again - I think doing it this way is fine, as is leaving the `.` in and using `??` (as above), but we need to be consistent. 

My preference is to keep it simple and use `??` as above, so that we don't need the extra `concatenateValueOrEmpty` function. 
```suggestion
            '.' . ($this->serviceAccountImpersonationUrl ?? '') .
            '.' . ($this->auth->getSubjectTokenType() ?? '') .
            '.' . ($this->workforcePoolUserProject ?? '');
```

Reply:
I did it to keep the `null` value in mind and not have a cache key like:
```
a.b...
```
We do remove the dots though, but still there was a previous comment about handling the nulls different and I decided to not add the random dots. What do you think?

Reply:
My point was that we want it to be consistent, as you were adding the dots already in AwsNativeSource